guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md

```diff
@@ -0,0 +1,146 @@
+---
+group: javascript-developer-guide
+subgroup: 3_Widgets
+title: ToggleAdvanced widget
+redirect_from:
+ - /guides/v2.0/frontend-dev-guide/javascript/widget_toggle.html
+---
+
+## Overview
+
+The ToggleAdvanced {% glossarytooltip f0dcf847-ce21-4b88-8b45-83e1cbf08100 %}widget{% endglossarytooltip %} is used to toggle HTML class for selectors by click on a toggle element.
+
+Also, it allows toggling a clicked element class and a clicked element text if an element is toggled.
+
+The ToggleAdvanced widget source is [lib/web/mage/toggle.js].
+
+## Initialize the ToggleAdvanced widget
+
+For information about how to initialize a widget in a JS component or `.phtml` template, see the [Initialize JavaScript] topic.
+
+Generally the ToggleAdvanced widget is instantiated like following:
+
+```javascript
+$("#element").toggleAdvanced();
+```
+
+Where:
+-   `#element` is the selector of the element for ToggleAdvanced is initialized.
+
+Phtml template file examples using script:
+
+```html
+<script>
+    require([
+        'jquery',
+        'toggleAdvanced'
+    ], function ($) {
+        'use strict';
+        
+        $("#element").toggleAdvanced();
+    });
+</script>
+```
+
+## Options
+
+The ToggleAdvanced widget has the following options:
+
+- [baseToggleClass](#basetoggleclass)
+- [selectorsToggleClass](#selectorstoggleclass)
+- [toggleContainers](#togglecontainers)
+- [newLabel](#newlabel)
+- [curLabel](#curlabel)
+- [currentLabelElement](#currentlabelelement)
+
+### `baseToggleClass`
+
+The class used to be toggled on clicked element.
+
+**Type**: String
+
+**Default value**: `active`
+
+### `selectorsToggleClass`
+
+The class used to be toggled on selectors DOM elements.
+
+**Type**: String
+
+**Default value**: `hidden`
+
+### `toggleContainers`
+
+The selectors for DOM elements we wish to toggle.
+
+**Type**: String
+
+**Default value**: `null`
+
+### `newLabel`
+
+Text of the new label to be used on toggle.
+
+**Type**: String
+
+**Default value**: `null`
+
+### `curLabel`
+
+Text of the old label to be used on toggle.
+
+**Type**: String
+
+**Default value**: `null`
+
+### `currentLabelElement`
+
+Current label of the container.
```

Comment:
I believe that it's rather a container of the current label element than the label of the container.
```suggestion
Container element of the current label
```